File: frontend/src/lib/api/browser-sessions.ts
import { postJson } from "@/lib/api/client";
import type {
  BrowserSessionCreateRequest,
  BrowserSessionCreateResponse,
} from "@/types/browser-session";


export function createBrowserSession(
  payload: BrowserSessionCreateRequest = {},
  accessToken?: string,
  signal?: AbortSignal,
) {
  return postJson<BrowserSessionCreateResponse, BrowserSessionCreateRequest>(
    "/api/v1/browser-sessions",
    payload,
    { accessToken, signal },
  );
}

File: backend/app/api/routes/browser_sessions.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.auth import get_current_user
from app.db.session import get_db
from app.schemas.auth import AuthenticatedUser
from app.schemas.browser_session import (
    BrowserSessionCreateRequest,
    BrowserSessionCreateResponse,
)
from app.services.browser_session_service import create_browser_session


router = APIRouter(prefix="/browser-sessions", tags=["browser-sessions"])


@router.post("", response_model=BrowserSessionCreateResponse)
def create_browser_session_route(
    payload: BrowserSessionCreateRequest,
    current_user: AuthenticatedUser = Depends(get_current_user),
    db: Session = Depends(get_db),
) -> BrowserSessionCreateResponse:
    return create_browser_session(db, payload, user_id=current_user.id)
